src/Orleans.Core.Abstractions/IDs/SiloAddress.cs

```diff
@@ -284,26 +279,16 @@ internal void InternalSetConsistentHashCode(int hashCode)
             this.hashCodeSet = true;
         }
 
-        // This is the same method as Utils.CalculateIdHash
-        private static int CalculateIdHash(string text)
+        internal static int CalculateIdHash(string text)
         {
-            SHA256 sha = SHA256.Create(); // This is one implementation of the abstract class SHA1.
-            int hash = 0;
-            try
-            {
-                byte[] data = Encoding.Unicode.GetBytes(text);
-                byte[] result = sha.ComputeHash(data);
-                for (int i = 0; i < result.Length; i += 4)
-                {
-                    int tmp = (result[i] << 24) | (result[i + 1] << 16) | (result[i + 2] << 8) | (result[i + 3]);
-                    hash = hash ^ tmp;
-                }
-            }
-            finally
-            {
-                sha.Dispose();
-            }
-            return hash;
+            var input = BitConverter.IsLittleEndian ? MemoryMarshal.AsBytes(text.AsSpan()) : Encoding.Unicode.GetBytes(text);
+
+            Span<int> result = stackalloc int[256 / 8 / sizeof(int)];
+            SHA256.HashData(input, MemoryMarshal.AsBytes(result));
```

Comment:
I'll look into this for a separate PR as this is used from multiple places. Probably could even calculate it based on the raw values (instead of first creating a string).